src/devicecontroller/DefaultDeviceController.ts

```diff
@@ -700,11 +716,33 @@ export default class DefaultDeviceController implements DeviceControllerBasedMed
       }
       newDevice.groupId = this.getGroupIdFromDeviceId(kind, this.getDeviceIdStr(device));
     } catch (error) {
+      let errorMessage: string;
+      if (error?.name && error.message) {
+        errorMessage = `${error.name}: ${error.message}`;
+      } else if (error?.name) {
+        errorMessage = error.name;
+      } else if (error?.message) {
+        errorMessage = error.message;
+      } else {
+        errorMessage = 'UnknownError';
+      }
+
+      if (kind === 'audio') {
+        this.boundAudioVideoController?.eventController.publishEvent('audioInputFailed', {
```

Comment:
Question: Would it be better if we have these `string` events attached to some enum since the `eventDidReceive` and `publishEvent` gets an event name based on `EventName` type? For example rather than 'audioInputFailed' we could have this governed by some string enum (not indexed) and use the same enum for `name` argument in the `eventDidReceive`?

Reply:
`publishEvent` takes in the `EventName` type that is a string union of `audioInputFailed` and other event strings, so you can't call `publishEvent` with a non-event string.

Also, I chose a string union type over a enum for two reasons.
1. A builder doesn't need to import additonal enum files for comparsion.
2. A builder doesn't have to know the enum value.

Example using enums
```
// 1. A JavaScript builder needs to import enum files.
import EventName from './EventName';
import EventAttributes from './EventAttributes';

eventDidReceive(name, attributes): void {
  if (name === EventName.AudioInputFailed) {
     // 2. Before sending this event to the analytics tool, a builder needs to check the value of 
           "EventName.AudioInputFailed" to make sure that it indicates the event name.   
```

Example using a string union type
```
eventDidReceive(name, attributes): void {
   if (name === 'audioInputFailed') {
       // They know the "name" argument is an event-name string.
}
```

Reply:
Cool, got it thanks for the explanation!